### ISSUE TITLE
cp -r behaves differently when target_dir is empty or not

### ISSUE THREAD

**Issue by therealchfkch** [nushell/nushell#13717]
### Describe the bug

When i use `cp -r source-dir target-dir` and `target-dir` does not exist yet, it copies the content (without the `source-dir` folder) to `target-dir.
On the other hand, when `target-dir` already exists, the result is `target-dir/source-dir` with the contents.

### How to reproduce

1. Create a directory (named `source-dir` for this test) and place one or more dummy files in it (`file1.md`, `file2.md`).
2. execute `cp -r source-dir target-dir`
3. make sure the files are directly located in the root of `target-dir`
4. execute `cp -r source-dir target-dir` again
5. now the files are in a subfolder of `target-dir` named `source-dir`

### Expected behavior

step 4 produces results like in step 2 & 3 as described above

OR

step 2 produces results like step 4 & 5 as described above

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.96.1                                                            |
| major              | 0                                                                 |
| minor              | 96                                                                |
| patch              | 1                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.79.0 (129f3b996 2024-06-10) (built from a source tarball) |
| cargo_version      | cargo 1.79.0                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash                          |
| installed_plugins  |                                                                   |

**Comment by therealchfkch** [comment]
# output after step 2: 
erd target-dir/                                                                                                                                                  
0 B ┌─ file1.md
0 B ├─ file2.md
 target-dir

2 files


# output after step 4:
0 B    ┌─ file1.md
0 B    ├─ file2.md
 ┌─ source-dir
0 B ├─ file1.md
0 B ├─ file2.md
 target-dir

1 directory, 4 files

**Comment by therealchfkch** [comment]
# Run with --debug and -v flag
## output of step 2
cp -r source-dir target-dir --debug -v                                                                                                                           
'/home/roman/source-dir' -> '/home/roman/target-dir/'
'/home/roman/source-dir/file1.md' -> '/home/roman/target-dir/file1.md'
copy offload: unknown, reflink: unsupported, sparse detection: no
'/home/roman/source-dir/file2.md' -> '/home/roman/target-dir/file2.md'
copy offload: unknown, reflink: unsupported, sparse detection: no

## output of step 4
cp -r source-dir target-dir --debug -v                                                                                                                           
'/home/roman/source-dir' -> '/home/roman/target-dir/source-dir'
'/home/roman/source-dir/file1.md' -> '/home/roman/target-dir/source-dir/file1.md'
copy offload: unknown, reflink: unsupported, sparse detection: no
'/home/roman/source-dir/file2.md' -> '/home/roman/target-dir/source-dir/file2.md'
copy offload: unknown, reflink: unsupported, sparse detection: no

**Comment by fdncred** [comment]
since we use uu_cp from uutils/coreutils it would be good to understand if this problem is upstream or in our code.

**Comment by therealchfkch** [comment]
> since we use uu_cp from uutils/coreutils it would be good to understand if this problem is upstream or in our code.

you are right, i just installed the uutils package and used it directly - with the same results. Should i open an issue there and close this or how do we proceed?

**Comment by fdncred** [comment]
ya, you should open an issue on their repo for uu_cp and reference it here and then we have to wait to find out if they decide to fix it.

**Comment by therealchfkch** [comment]
Upstream uutils/coreutils issue https://github.com/uutils/coreutils/issues/6671
